netket/experimental/sampler/metropolis_pt.py

```diff
@@ -161,16 +182,85 @@ def __init__(self, *args, n_replicas: int = 32, **kwargs):
+            if isinstance(betas, str):
+                beta_distribution = betas
+                if not (beta_distribution == "lin" or beta_distribution == "log"):
+                    raise ValueError(
+                        f"""To initialize the temperatures with a string, you must choose between "lin" for a linear distribution and "log" for a logarithmic distribution. Instead got "{beta_distribution}"."""
+                    )
+
+                # default if not provided
+                if n_replicas is None:
+                    n_replicas = 32
+
+                # list is empty if we have a string
+                betas = None
+
+            # or we have the list
+            else:
+                betas = jnp.array(betas, dtype=jnp.float64).reshape(-1)
+
+                # verify that the list is compatible with n_replicas
+                if n_replicas is not None and betas.shape[-1] != n_replicas:
+                    raise ValueError(
+                        f"""The dimension of the array beta and the number of replicas must match, instead, got {n_replicas} replicas but {betas.shape[-1]} values for beta."""
+                    )
+
+                # is the list valid
+                if not (jnp.isclose(jnp.max(betas), 1) and jnp.min(betas) > 0):
+                    raise ValueError(
+                        rf"""The values for beta should be in (0,1] and obligatory contain beta=1, instead got [{jnp.min(betas):.2f},{jnp.max(betas):.8f}]."""
+                    )
+
+                # sort and adapt the rest correspondingly
+                betas = jnp.sort(betas, descending=True)  # we need beta[0] = 1
+                n_replicas = betas.shape[-1]
+                beta_distribution = "custom"
+
+        # verify the number of replicas
         if not (
             isinstance(n_replicas, int)
             and n_replicas > 0
             and np.mod(n_replicas, 2) == 0
         ):
             raise ValueError("n_replicas must be an even integer > 0.")
+
         self.n_replicas = n_replicas
+        self._beta_sorted = betas
+        self._beta_distribution = beta_distribution
 
         super().__init__(*args, **kwargs)
 
+    @property
+    def sorted_betas(self):
+        """
+        The sorted values of the temperatures for each _physical_ markov chain.
+        The first value is β = 1 and is the _physical_ temperature.
+        """
+
+        if self._beta_sorted is not None:
+            return self._beta_sorted
+        else:
+            if self._beta_distribution == "lin":
+                return (
+                    1.0
+                    - jnp.arange(self.n_replicas, dtype=jnp.float64) / self.n_replicas
+                )
+            elif self._beta_distribution == "log":
+                return -jnp.log(
+                    jnp.arange(1, self.n_replicas + 1, dtype=jnp.float64)
+                    / (self.n_replicas + 1)
+                ) / jnp.log(self.n_replicas + 1)
+
```

Comment:
just add an error check here to be sure...
```suggestion
            else:
                raise NotImplementedError(f"distribution: {self._beta_distribution}")

```